storage/myisam/mi_write.c

```diff
@@ -926,7 +926,7 @@ static int keys_compare(bulk_insert_param *param, uchar *key1, uchar *key2)
 }
 
 
-static int keys_free(uchar *key, TREE_FREE mode, bulk_insert_param *param)
+static void keys_free(void* vkey, TREE_FREE mode, void *vparam)
```

Comment:
Please, also report this to upstream. I still think that calling a int-returning function cast to a void-returning function can potentially lead to code corruption.

Reply:
Reported at https://bugs.mysql.com/bug.php?id=90412